src/services/books.ts

```diff
@@ -1,4 +1,12 @@
 import api from '../config/api';
-import { BookListResponse } from '../types/book';
+import { BookListResponse, Book } from '../types/book';
 
 export const getBooks = () => api.get<BookListResponse>('/books');
+
+export const getBookDetail = async (id: string): Promise<Book> => {
```

Comment:
react-query resolves promises, using this like this is like you're not properly using react-query right

 query resexport const getBook = (id: string) => headers.get<Book>(`/books/${id}`, {});
```suggestion
export const getBookDetail = (id: string) => api.get<Book>(`/books/${id}`, {});
```

Reply:
I'm trying to use react query like the first getBooks but always get an error with the types in the useQuery call, this is an example that I try but I always get an error
<img width="712" alt="Screen Shot 2022-03-17 at 10 02 51 AM" src="https://user-images.githubusercontent.com/32044536/158851621-52474fcc-74ad-4300-9052-3b2f84b890dc.png">
<img width="653" alt="Screen Shot 2022-03-17 at 10 03 43 AM" src="https://user-images.githubusercontent.com/32044536/158851638-b24e2944-2dd6-4e00-9974-b2b31a4b16a5.png">
.

Reply:
`if you can try const { isLoading, data, isError } = useQuery(['bookDetail', id], () => getBookDetail(id));`

in this file src/screens/BookDetail/index.tsx